### ISSUE TITLE
Accessing Sub Aggregation Results?

### ISSUE THREAD

**Issue by kylebrandt** [olivere/elastic#10]
With the following:

```
    q := elastic.NewMatchQuery("message", "error")
    dh := elastic.NewDateHistogramAggregation().Field("@timestamp").Interval("5m")
    t := elastic.NewTermsAggregation().Field("logsource").SubAggregation("ts", dh)
    result, err := client.Search().Index("logstash-2014.11.07").Query(q).Aggregation("groups", t).Do()
```

I'm not really sure how to access the sub aggregations.

`fmt.Println(string(result.Aggregations["groups"]))` returns `{"buckets":[{"key":"ny","doc_count":200851,"ts":{"buckets":[{"key_as_string":"2014-11-07T00:00:00.000Z","key":1415318400000,"doc_count":879}`

So the sub aggregation seems to be working, however:

```
    if groups, ok := result.GetAggregation("groups"); ok {
        _, hmm := groups.GetAggregation("ts")
        fmt.Println(hmm)
    }
```

returns false

**Comment by justinfx** [comment]
I came across the issue after spending hours trying to figure out why I could not access the sub-aggregations in my query. I had checked the raw query being generated, tried all kinds of variations of structuring the query builder, adjusted the scope of variables in inner loops... Then I saw that this issue illustrates that the sub-aggregations have to be access on the bucket object and I was accidentally trying to look them up by the parent aggregation (`AggregationBucketKeyItems`). This was always returning false.

I was doing this:

```
groups, _ := result.Aggregations.Terms("groups")
for _, b := range groups.Buckets {
  ts, _ := groups.DateHistogram("ts")
  for _, dhb := range ts.Buckets {
    // ...
  }
}
```

When I should have been doing this:

```
groups, _ := result.Aggregations.Terms("groups")
for _, b := range groups.Buckets {
  ts, _ := b.DateHistogram("ts")
  for _, dhb := range ts.Buckets {
    // ...
  }
}
```

The mistake is dumb on my part, now that I realise it. But the problem is that both the aggregation object and the bucket have a similar interface so I was able to call `Terms()` and not see a difference.
 
Would it be possible to add an explicit example of reading the results of sub-aggregations to your Wiki? 
https://github.com/olivere/elastic/wiki/QueryDSL

It is very difficult to search through the massive amount of test code to try and see how it is done :-/
